controllers/prep_handlers.go

```diff
@@ -128,6 +131,35 @@ func (r *ImageBasedUpgradeReconciler) checkSeedImageCompatibility(_ context.Cont
 	return nil
 }
 
+// validateSeedOcpVersion block with an error if seed image version is less or equal than current cluster (target) OCP version
+func (r *ImageBasedUpgradeReconciler) validateSeedOcpVersion(seedOcpVersion string) error {
+	// get target OCP version
+	targetClusterVersion := &configv1.ClusterVersion{}
+	if err := r.Get(context.Background(), types.NamespacedName{Name: "version"}, targetClusterVersion); err != nil {
+		return fmt.Errorf("failed to get ClusterVersion for target: %w", err)
+	}
+	targetOCP := targetClusterVersion.Status.Desired.Version
+
+	// parse versions
+	targetSemVer, err := semver.NewVersion(targetOCP)
+	if err != nil {
+		return fmt.Errorf("failed to parse target version %s: %w", targetOCP, err)
+	}
+	seedSemVer, err := semver.NewVersion(seedOcpVersion)
+	if err != nil {
+		return fmt.Errorf("failed to parse seed version %s: %w", seedOcpVersion, err)
+	}
+
+	// compare versions
+	if seedSemVer.Equal(*targetSemVer) || seedSemVer.LessThan(*targetSemVer) {
```

Comment:
I think this could just be:
```suggestion
	if seedSemVer.Compare(*targetSemVer) <= 0 {
```